### PR TITLE
Uniform lerp/slerp function for vectors and quaterion

### PR DESCRIPTION
**Pull request by shrinktofit**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by stanleyljl** [comment]
@stanleyljl test

**Comment by stanleyljl** [comment]
@stanleyljl ttt

**Review by stanleyljl — APPROVED** [review]

File: native/cocos/math/Quaternion.h [hunk 1]
```diff
@@ -432,6 +441,24 @@ class CC_DLL Quaternion {
...
 };
 
+/**
+ * Quaterion linear interpolation.
+ */
+inline Quaternion lerp(const Quaternion& from, const Quaternion& to, Quaternion::ValueType t) {
+    Quaternion result;
+    Quaternion::lerp(from, to, t, &result);
+    return result;
+}
```
Comment (minggo) [inline native/cocos/math/Quaternion.h:451]: It is not a good idea to implement it in cocos name space. It is easy to cause conflict. It is better to implement as a static function.

File: native/cocos/math/Quaternion.h [hunk 2]
```diff
@@ -432,6 +441,24 @@ class CC_DLL Quaternion {
...
+
+/**
+ * Quaterion spherical interpolation.
+ */
+inline Quaternion slerp(const Quaternion &from, const Quaternion &to, Quaternion::ValueType t) {
+    Quaternion result;
+    Quaternion::slerp(from, to, t, &result);
+    return result;
+}
```
Comment (minggo) [inline native/cocos/math/Quaternion.h:460]: The same as above.

File: native/cocos/math/Quaternion.h (hunk 1 above)
Comment (minggo) [reply native/cocos/math/Quaternion.h:451]: More better is to implement as a member function.

**Comment by minggo** [comment]
@stanleyljl please take a look.

**Comment by shrinktofit** [comment]
Shoul have been replaced by other PRs.

**Pull request closed without merging**